src/pages/register.tsx

```diff
@@ -45,42 +47,47 @@ export default function UserRegistration({ data }) {
     } else {
       setErrorMessage(message);
     }
+    showModalHandler();
   };
 
   const resetForm = () => {
     setErrorMessage("");
     setStatus(FormStatus.Unsubmitted);
   };
 
+  const showModalHandler = () => {
+    showModal({
+      title: `Whoops!`,
+      body: (
+        <>
+          <p>An error occurred while processing your registration.</p>
+          {errorMessage !== "" && (
+            <p>
+              <small>{errorMessage}</small>
+            </p>
+          )}
+        </>
+      ),
+    });
+  };
```

Comment:
Apologies - it seems we had a miscommunication about this. We should actually show the error message on the page, like we do with error messages on other forms, not in a modal.

The issue with the form state resetting when an error occurs is due to the fact that the error state is managed outside of the form component. The `register` page component manages the error state, so when an error occurs, the `RegisterForm` component is unmounted and the error message is shown instead. Then, when the error state is reset and we show the form again, it is actually remounting the form component with blank state. If we either 1. manage error state inside `RegisterForm`, or 2. remove `RegisterForm` altogether and move its contents directly into `register` page component, we can solve this problem.